expyfun/stimuli/_tracker.py

```diff
@@ -0,0 +1,837 @@
+
+    @property
+    def step_size_down(self):
+        return self._step_size_down
+
+    @property
+    def stop_criterion(self):
+        return self._stop_criterion
+
+    @property
+    def stop_rule(self):
+        return self._stop_rule
+
+    @property
+    def start_value(self):
+        return self._start_value
+
+    @property
+    def x_min(self):
+        return self._x_min
+
+    @property
+    def x_max(self):
+        return self._x_max
+
+    @property
+    def stopped(self):
+        """Has the tracker stopped
+        """
+        return self._stopped
+
+    @property
+    def x(self):
+        """The staircase
+        """
+        return self._x
+
+    @property
+    def x_current(self):
+        """The current level
+        """
+        return self._x_current
+
+    @property
+    def responses(self):
+        """The response history
+        """
+        return self._responses
+
+    @property
+    def n_trials(self):
+        """The number of trials so far
+        """
+        return self._n_trials
+
+    @property
+    def n_reversals(self):
+        """The number of reversals so far
+        """
+        return self._n_reversals
+
+    @property
+    def reversals(self):
+        """The reversal history (0 where there was no reversal)
+        """
+        return self._reversals
+
+    @property
+    def reversal_inds(self):
+        """The trial indices which had reversals"""
+        return np.where(self._reversals)[0]
+
+    # =========================================================================
+    # Display functions
+    # =========================================================================
+    def plot(self):
+        """Plot the adaptive track.
+
+        Returns
+        -------
+        fig : Figure
+            The figure handle.
+        ax : AxesSubplot
+            The axes handle.
+        lines : list of Line2D
+            The handles to the staircase line and the reversal dots.
+        """
+        fig, ax = plt.subplots(1)
+
+        line = ax.plot(1 + np.arange(self._n_trials), self._x, 'k.-')
+        dots = ax.plot(1 + np.where(self._reversals > 0)[0],
+                       self._x[self._reversals > 0], 'ro')
+        ax.set(xlabel='Trial number', ylabel='Level')
+        return fig, ax, line + dots
+
+    def plot_thresh(self, n_skip=2, ax=None):
+        """Plot a line showing the threshold.
+
+        Parameters
+        ----------
+            n_skip : int
+                See documentation for ``TrackerUD.threshold``.
+            ax : Axes
+                The handle to the axes object. If None, the current axes will
+                be used.
+
+        Returns
+        -------
+        line : list Line2D
+            The handle to the threshold line, as returned from ``plt.plot``.
+        """
+        if ax is None:
+            ax = plt.gca()
+        h = ax.plot([1, self._n_trials], [self.threshold(n_skip)] * 2,
+                    '--', color='gray')
```

Comment:
that is true of the `plot()` method but not the `plot_thresh()` method.  But you raise a good point that it would be hard to do cleanly for the `plot()` method, so for consistency I'll agree to leave both methods as-is (with hard-coded styles).